### PR TITLE
Update supported React version (>= 16)

### LINKED ISSUE #75: Add support for React 17
Currently peer dependencies are locked to 16.6 version. Please update to support 17.

### PR DESCRIPTION
**Pull request by andyrichardson**
Fix #75 

Had a brief look at the code and I'm not seeing anything that would be impacted by React 17

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "release-it": "^12.4.1"
   },
   "peerDependencies": {
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0"
+    "react": ">=16.0.0",
+    "react-dom": ">=16.0.0"
```
Comment (oliviertassinari) [inline package.json:64]: What if React 18 introduces a breaking change?

Comment (andyrichardson) [reply package.json:64]: What if React 18 doesn't introduce a breaking change?

I kid 😅  I know there are differing opinions on this but IMHO, keeping this open is more useful.

Anyone updating their app to a new major release of React is (I would hope) testing for breakages in their app. By creating a constraint ahead of time, we prevent users from making that upgrade until confirmation for this package is made (that confirmation coming in the form of a new release with the dependency constraint updated).

Yarn resolutions can work around this but that only applies to folks using `yarn` and not `npm`.

**Review by crossjs — APPROVED** [review]
would be nice

**Comment by FDiskas** [comment]
How long we going to wait?

**Review by ext — APPROVED** [review]

**Comment by astutejoe** [comment]
That would be really nice to see! :)

**Comment by FDiskas** [comment]
I landed here because I'm using styleguidist and it has this as dependency.

**Comment by ext** [comment]
> I landed here because I'm using styleguidist and it has this as dependency.

Me too, and worse is that it wont work properly with NPM 7 as it pulls two conflicting versions of react unless using `--legacy-peer-deps`

**Comment by jrkienle** [comment]
@satya164 Can we get an update on if this is going to be approved and released or not? This module being a key dependency of `react-styleguidist` makes these changes important to a lot of people

File: package.json (hunk 1 above)
Comment (rgischk) [reply package.json:64]: Could this work? Would prevent React 18 from breaking something once it releases.
```suggestion
    "react": "^16.0.0 || ^17.0.0",
    "react-dom": "^16.0.0 || ^17.0.0"
```

**Comment by satya164** [comment]
Thanks for the PR! I'm closing this since this is now fixed in main.

**Pull request closed without merging**